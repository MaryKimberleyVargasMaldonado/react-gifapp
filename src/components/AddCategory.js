import React, { useState } from "react";
import { ReactDOM } from "react";
import PropTypes from "prop-types";

export const AddCategory = (props) => {
	const [inputValue, setInputValue] = useState("Search a category:");

	// H A N D L E   I N P U T   C H A N G E
	const handleInputChange = (e) => {
		setInputValue(e.target.value);
		console.log("HandleInputChange llamado.");
	};

	//colocar el event a un form
	// H A N D L E   S U B M I T
	const handleSubmit = (e) => {
		e.preventDefault(); //prevenir comportamiento por defecto del formulario.
		console.log("Submit done", inputValue);

		if (inputValue.trim().length > 2) {
			props.setCategories((cate) => [inputValue, ...cate]);
			setInputValue("");
		}
	};

	return (
		//FRAGMENT
		<form onSubmit={handleSubmit}>
			<p>{inputValue}</p>
			<h2>{inputValue}</h2>
			<input type="text" value={inputValue} onChange={handleInputChange} />
		</form>
	);
};

// P R O P E R T I E S
AddCategory.propTypes = {
	setCategories: PropTypes.func.isRequired,
};
