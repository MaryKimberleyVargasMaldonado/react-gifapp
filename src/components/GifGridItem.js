import React from "react";
//import { GifGrid } from "./components/GiftGrid";
import PropTypes from "prop-types";

export const GifGridItem = ({ title, url }) => {
	console.log({ title, url });
	return (
		<div className="card animate__animated animate__zoomInDown">
			<img src={url} alt={title} />
			<p>{title}</p>
		</div>
	);
};

// P R O P E R T I E S   -   P R O P T Y P E S
GifGridItem.propTypes = {
	title: PropTypes.string.isRequired,
	url: PropTypes.string.isRequired,
	//id: PropTypes.string.isRequired,
};
