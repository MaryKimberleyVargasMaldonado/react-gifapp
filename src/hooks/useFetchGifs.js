import { useEffect, useState } from "react/cjs/react.development";
import { getGifs } from "../helpers/getGifs";

export const useFetchGifs = (category) => {
	const [state, setState] = useState({
		data: [],
		loading: true,
	});

	// U S E   E F F E C T  con lista de dependencias
	useEffect(() => {
		return getGifs(category).then((imgs) => {
			setState({ data: imgs, loading: false });
		});
	}, [category]); // para que sólo se dispare una única vez

	return state; // {data:[], loading:true}
};
/*
	setTimeout(() => {
		setState({
			data: [1, 2, 3, 4, 5, 6, 7],
			loading: false,
		});
	});
*/
