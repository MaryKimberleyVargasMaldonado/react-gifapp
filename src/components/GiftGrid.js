import React from "react";
import { useState, useEffect, Fragment } from "react/cjs/react.development";
//import { getGifs } from "../helpers/getGifs";
import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifGridItem } from "./GifGridItem";

export const GifGrid = ({ category }) => {
	// U S E   F E T C H   G I F T S
	const { data: images, loading } = useFetchGifs(category);

	// I M A G E S
	/*
	const [images, setImages] = useState([]);


*/
	//getGifs();
	return (
		<Fragment>
			<h3 className="animate__animated animate__fadeIn">{category}</h3>
			{loading && <p className="animate__animate animate__flash">Loading...</p>}
			<div className="card-grid">
				{images.map((img) => (
					<GifGridItem key={img.id} {...img} />
				))}
			</div>
		</Fragment>
	);
};

/*
			<ol>
				{images.map((img) => (
					<li key={img.id}>{img.title}</li>
				))}
			</ol>*/
