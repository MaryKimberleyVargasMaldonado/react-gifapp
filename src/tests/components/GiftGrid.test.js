import { GifGrid } from "../../components/GiftGrid";
import React from "react";
import { shallow } from "enzyme";
import "@testing-library/jest-dom";
import { useFetchGifs } from "../../hooks/useFetchGifs";
import { GifGridItem } from "../../components/GifGridItem";
jest.mock("../../hooks/useFetchGifs");

describe("Prueba en GiftGrid.js", () => {
	const category = "Husky";

	// P R I M E R   T E S T
	test("1. Snapshot :", () => {
		useFetchGifs.mockReturnValue({ data: [], loading: true });
		const wrapper = shallow(<GifGrid category={category} />);
		expect(wrapper).toMatchSnapshot();
	});

	// S E G U N D O   T E S T
	test("2. Mostrar items cuando cargan imÃ¡genes en useFetchGifs: ", () => {
		const gifs = [
			{
				id: "HUMB",
				url: "https://localhost/cualquier.jpg",
				title: "cualquier",
			},
		];
		useFetchGifs.mockReturnValue({ data: gifs, loading: false });

		const wrapper = shallow(<GifGrid category={category} />);
		expect(wrapper).toMatchSnapshot();
		expect(wrapper.find("p").exists()).toBe(false);
		expect(wrapper.find("GifGridItem").length).toBe(gifs.length);
	});

	// T E R C E R   T E S T
	test("3. Exist: ", () => {});
});
