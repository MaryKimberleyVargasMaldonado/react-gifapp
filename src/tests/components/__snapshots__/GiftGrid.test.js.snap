// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en GiftGrid.js 1. Snapshot : 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Husky
  </h3>
  <p
    className="animate__animate animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Prueba en GiftGrid.js 2. Mostrar items cuando cargan im√°genes en useFetchGifs:  1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Husky
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="HUMB"
      key="HUMB"
      title="cualquier"
      url="https://localhost/cualquier.jpg"
    />
  </div>
</Fragment>
`;
